cli/run_util.go

```diff
@@ -566,103 +566,126 @@ func (r WorkflowRunner) runStep(
 	return 0, updatedStepOutputs, nil
 }
 
-type DockerManager interface {
-	Login(models.Container, map[string]string) error
-	StartWorkflowContainer(models.Container, string, map[string]string) (*docker.RunningContainer, error)
-	StartServiceContainers(services map[string]models.Container, workflowID string, envs map[string]string) ([]*docker.RunningContainer, error)
-	GetWorkflowContainer(string) *docker.RunningContainer
-	GetServiceContainers(string) []*docker.RunningContainer
-	DestroyAllContainers() error
+func (r WorkflowRunner) startContainersForStepGroup(containerID string, serviceIDs []string, environments *[]envmanModels.EnvironmentItemModel, groupID, workflowTitle string) {
+	if containerID == "" && len(serviceIDs) == 0 {
+		return
+	}
+
+	if err := tools.EnvmanInit(configs.InputEnvstorePath, true); err != nil {
+		log.Debugf("Couldn't initialize envman.")
+	}
+	if err := tools.EnvmanAddEnvs(configs.InputEnvstorePath, *environments); err != nil {
+		log.Debugf("Couldn't add envs.")
+	}
+
+	envList, err := tools.EnvmanReadEnvList(configs.InputEnvstorePath)
+	if err != nil {
+		log.Debugf("Couldn't read envs from envman.")
+	}
+
+	if containerID != "" {
+		containerDef := r.ContainerDefinition(containerID)
+		if containerDef != nil {
+			log.Infof("ℹ️ Running workflow in docker container: %s", containerDef.Image)
+
+			// TODO: Why only login for step execution containers?
```

Comment:
I think the original assumption was that services containers are public utility images like `postgres`. But it should also support authorization I think